### PR TITLE
Support canonicalization for Compute network values

### PR DESCRIPTION
**Pull request by maqiuyujoyce**
### BRIEF Change description

- Added `CanonicalizeAndNormalize(ctx, reader, defaultNamespace, parentProjectID, projectMapper)` method to `ComputeNetworkRef` to combine network canonicalization (expanding short names, stripping URI prefixes, and converting project numbers) and Kubernetes reference normalization into a single call.
- Made the internal helper `canonicalizeNetworkValue` unexported.
- Added comprehensive unit tests in `computenetwork_reference_test.go` covering `canonicalizeNetworkValue` edge cases and `CanonicalizeAndNormalize` round-trip resolutions.

#### Does this PR add something which needs to be 'release noted'?

```release-note
NONE
```

- [ ] Reviewer reviewed release note.

#### Additional documentation e.g., references, usage docs, etc.:

```docs

```

#### Intended Milestone

Please indicate the intended milestone. 
- [ ] Reviewer tagged PR with the actual milestone.

### Tests you have done

- [ ] Run `make ready-pr` to ensure this PR is ready for review.
- [ ] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: apis/compute/refs/computenetwork_reference.go [hunk 1]
```diff
@@ -45,6 +46,10 @@ var _ refsv1beta1.Ref = &ComputeNetworkRef{}
 // ComputeNetworkRef is a reference to a GCP ComputeNetwork.
 // +k8s:deepcopy-gen=true
 type ComputeNetworkRef struct {
+	// For backward compatibility, the external value can also be full URIs
+	// (e.g. "https://www.googleapis.com/compute/v1/projects/{{projectID}}/global/networks/{{networkID}}")
+	// or short names (e.g. "my-vpc").
```
Comment (gemmahou) [inline apis/compute/refs/computenetwork_reference.go:51]: I feel it worths make this message visible to user, can we update the the "Should be in the format" below to include all the three valid formats?

File: apis/compute/refs/computenetwork_identity.go [hunk 2]
```diff
@@ -39,6 +39,10 @@ func (i *ComputeNetworkIdentity) String() string {
 	return ComputeNetworkIdentityFormat.ToString(*i)
 }
 
+// FromExternal parses ref into a ComputeNetworkIdentity.
+// It supports full HTTPS URIs ("https://www.googleapis.com/compute/v1/projects/{project}/global/networks/{network}")
+// and relative resource names ("projects/{project}/global/networks/{network}").
+// Raw short names (e.g. "my-vpc") are not valid external references until canonicalized via CanonicalizeExternal.
```
Comment (gemmahou) [inline apis/compute/refs/computenetwork_identity.go:45]: Question: why do we handle short names separately? If the goal is backward capability, we should support both full url and short names?

File: apis/compute/refs/computenetwork_reference.go (hunk 1 above)
Comment (maqiuyujoyce) [reply apis/compute/refs/computenetwork_reference.go:51]: The reason we don't explicitly mention all three formats is that we don't encourage any format other than the relative resource name. I think promoting the consistent format is what KCC, and probably even GCP wants to do, so I prefer we don't make the other options visible to users. What do you think?

File: apis/compute/refs/computenetwork_identity.go (hunk 2 above)
Comment (maqiuyujoyce) [reply apis/compute/refs/computenetwork_identity.go:45]: Full HTTPS URIs already include the project ID path segment (`projects/{project}/...`), so `FromExternal` can parse them deterministically without external context. In contrast, raw short names (e.g. `my-vpc`) lack project information, so they cannot be parsed into an identity until `CanonicalizeExternal` inspects the environment/parent project to infer the missing project.

**Review by gemmahou — APPROVED** [review]
